### PR TITLE
Add a separate analysis pass

### LINKED ISSUE #252: Incorrect trait name in doc symbol dictionary
```patch
-To set the value of a child property, use `ContainerExt::child_set_property`,
+To set the value of a child property, use `Container::child_set_property`,
 `ContainerExt::child_set` or `ContainerExt::child_set_valist`.
 To obtain the value of a child property, use
-`ContainerExt::child_get_property`, `ContainerExt::child_get` or
+`Container::child_get_property`, `ContainerExt::child_get` or
```

### PR DESCRIPTION
**Pull request by gkoz**
Objects (classes and interfaces) and records ('generated' and 'manual' ones from all namespaces) get analyzed prior to generation. The `Info`s are collected into vectors with no indexing by name yet.
This partially implements #136. Fixes #252.

### CODE REVIEW

**Comment by EPashkin** [comment]
:+1:  Thanks

File: src/codegen/doc/mod.rs [hunk 1]
```diff
@@ -46,41 +47,49 @@ pub fn generate(env: &Env) {
 
 fn generate_doc(mut w: &mut Write, env: &Env) -> Result<()> {
     try!(write_file_name(w, None));
+    let mut generators: Vec<(&str, Box<Fn(&mut Write, &Env) -> Result<()>>)> = Vec::new();
+
+    let ignored = |info: &InfoBase| {
+        info.type_id.ns_id != MAIN ||
+            env.config.objects.get(&info.full_name).map_or(true, |obj| obj.status.ignored())
```
Comment (EPashkin) [inline src/codegen/doc/mod.rs]: only needed `!= MAIN`

Comment (gkoz) [inline src/codegen/doc/mod.rs]: I'm assuming we don't want to emit docs for items which are ignored and thus absent from the crate.

Comment (EPashkin) [inline src/codegen/doc/mod.rs]: This items ignored on analysis phase and not get to `objects` or `records`

Comment (gkoz) [inline src/codegen/doc/mod.rs]: Yeah, good point :)

**Pull request merged**